osquery/tables/system/user_groups.h

```diff
@@ -50,6 +54,16 @@ struct user_t {
 template <typename uid_type, typename gid_type>
 static void getGroupsForUser(QueryData &results,
                              const user_t<uid_type, gid_type> &user) {
+#ifdef __APPLE__
+  int ngroups = getgroupcount(user.name, user.gid);
```

Comment:
I don't think that can happen in normal circumstances and the code works fine if it does happen, so I think that's unnecessary. I don't feel particularly strongly though.